tasks/ssh_private_key.yaml

```diff
@@ -0,0 +1,18 @@
+- name: ensure ssh private key is set
+  loop: |
+    {%- set private_keys = [] -%}
+    {%- for user_name, user_value in system.users.items() -%}
+      {%- if user_value.private_key is defined -%}
+        {%- set _ = private_keys.append({'user': user_name, 'key': user_value.private_key}) -%}
+      {%- endif -%}
+    {%- endfor -%}
+    {{ private_keys }}
+  loop_control:
+    label: "{{ item.user }}"
+  copy:
+    dest: "~{{ item.user }}/.ssh/id_ed25519"
+    content: "{{ item.key }}"
```

Comment:
Maybe use YAML block syntax here to ensure line feed at EOF. (Some implementations return an error when reading a key and there is no trailing line feed.)

```yaml
copy:
  content: |
    {{ item.key }}
```